### DISCUSSION TITLE
Use Bootstrap's Utility API from my scss file

### DISCUSSION THREAD

**Discussion by lonix1** [twbs/bootstrap#36438] [Q&A] [answered]
The Bootstrap docs explain how to [enable](https://getbootstrap.com/docs/5.1/utilities/spacing/#negative-margin) negative margins - but that is for the case where one customises Bootstrap.

I'm using it from a CDN, but I still want those classes (e.g. `mt-n1`). I'd like to import them via a mixin (or something like that) into my scss. The ["Utility API"](https://getbootstrap.com/docs/5.1/utilities/spacing/#utilities-api) shows that is possible, but I'm unsure how to invoke it from my scss file.

How do I do that?

For example:
````scss
// import bootstrap
@import "~bootstrap/scss/functions";
@import "~bootstrap/scss/variables";
@import "~bootstrap/scss/mixins";
@import "~bootstrap/scss/utilities";

// generate negative margin classes from $utilities:"negative-margin"
// ... <---what goes here?
````

I tried `@include generate-utility("negative-margin")` but that fails.

**Comment by mdo** [comment, marked as answer]
This docs page likely has more information for you on how to use the utility API: https://getbootstrap.com/docs/5.2/utilities/api/. In particular, there's a section dedicated to using the API that shows some helpful code snippets.

For the latest examples, check our staging site: https://twbs-bootstrap.netlify.app/docs/5.2/utilities/api/#using-the-api. We fixed a couple code snippets errors there that will come out when v5.2 goes stable.
